spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc

```diff
@@ -33,25 +33,32 @@ For instance, in the following example, `MyLogInterceptor` will be invoked first
 [source, java]
 ----
 @Retry
-@MyLog 
+@MyLog
 @MyPrint
-public void myInvok() {
-
- //do something
+public void myInvoke() {
+  // do something
 }
+
 @Priority(3000)
 @MyLog
-public class MyLogInterceptor{
- //do logging
+public class MyLogInterceptor {
+  // do logging
 }
 
 @Priority(5000)
 @MyPrint
-public class MyPrintInterceptor{
- //do printing
+public class MyPrintInterceptor {
+  // do printing
 }
 ----
 
+=== Treating custom throwables
 
+Throwing custom throwables from business methods annotated with any of the Fault Tolerance interceptor bindings results in non-portable behavior.
+The term "custom throwable" means: any class that is a subtype of `Throwable`, but isn't a subtype of `Error` or `Exception`.
+This includes `Throwable` itself, and _doesn't_ include `Error` and `Exception`.
 
-
+Some Fault Tolerance annotations allow configuring a set of exception types for various purposes.
+For example, `@Retry` includes the `retryOn` attribute which configures the set of exceptions on which retry will be performed.
+In any of these cases, specifying a custom throwable is always ignored.
```

Comment:
It seems a bit odd to say that throwing a custom `Throwable` results in undefined behaviour, but listing a custom `Throwable` in an annotation parameter results in it being ignored.

My initial thought was that we should a TCK to check that the custom throwable is ignored, but it's hard to write a good test when throwing the custom throwable results in undefined behaviour.

The only test I can think of that we could actually write would be to have something like `@Retry(retryOn=Throwable.class)` and to test that with both an `Exception` and an `Error`.

However, this spec now says that `Throwable.class` should be _ignored_ in the `retryOn` parameter, which means that in this test _nothing_ should be retried, which itself seems highly unintuitive.

I suggest that we don't need to say anything about the types listed in annotation parameters, and merely state that "Throwing a `Throwable` which is not assignable to either `Exception` or `Error` results in undefined behaviour."

Reply:
So, I did think about this, and my initial decision was to stick to current `@Retry` behavior. Because `@Retry` already specifies that custom throwables are ignored in `retryOn` and `abortOn`. Here, I just changed `@CircuitBreaker` to follow suit. We can do the opposite -- change `@Retry` to be the same as `@CircuitBreaker`, but IMHO we should align them one way or the other.

Reply:
Ah sorry, I misread the diff and thought you'd added that part for `Retry`. You can ignore this comment in that case.

I agree that they should at least be consistent, whichever behaviour we decide on.

I think this is worth more discussion though. With the current wording, I think the only possible test is the one that I outlined and the specified result is (at least to me) unexpected.

Reply:
Sure, let's discuss this on the call tomorrow.